felix/bpf/bpf.go

```diff
@@ -1311,8 +1311,10 @@ func CidrToHex(cidr string) ([]string, error) {
 		return nil, fmt.Errorf("IP %q is not IPv4", ip)
 	}
 
-	if mask > math.MaxUint32 || mask < 0 {
-		return nil, fmt.Errorf("mask %d should be between 0 and 4,294,967,295", mask)
+	// Check bounds on the mask against an int32 instead of uint32 since the mask value
+	// will be in subnet mask notation and should range between 0 and 32
+	if mask > math.MaxInt32 || mask < 0 {
```

Comment:
I could change this check to check against `32` instead of `math.MaxInt32` but figured that since this is an integer overflow check that I would make it more obvious as opposed to making this a validation of proper CIDR input.

Reply:
I don't fully follow, if this is subnet mask notation, the max (255.255.255.255) would be `MaxUint32`, not `MaxInt32` (4294967295)... Was there an issue with using the unsigned max?